src/python/grpcio/grpc/_importer.py

```diff
@@ -0,0 +1,190 @@
+                del path
+                del target
+                filepath = _module_name_to_proto_file(fullname)
+                for search_path in sys.path:
+                    try:
+                        prospective_path = os.path.join(search_path, filepath)
+                        os.stat(prospective_path)
+                    except (FileNotFoundError, NotADirectoryError):
+                        continue
+                    else:
+                        return importlib.machinery.ModuleSpec(
+                            fullname, ProtoLoader(fullname, filepath))
+
+        def services(protobuf_path, *, include_paths=None):
+            """Loads gRPC service classes and functions from a .proto file.
+
+            THIS IS AN EXPERIMENTAL API.
+
+            Returns a module object equivalent to a "_pb2_grpc.py" file.
+
+            Together with the google.protobuf.protos function, this enables
+            deployment of services without a code generation step beforehand.
+
+            Args:
+              protobuf_path: A string representing the path to the desired
+                ".proto" file.
+              include_paths: A sequence of strings which should be searched for
+                the ".proto" file. By default, the entries on sys.path are
+                searched.
+
+            Returns:
+              A module object containing servicers, stubs, and associated
+              functions for the requested ".proto" file.
+            """
+            with _augmented_syspath(include_paths):
+                module_name = _proto_file_to_module_name(protobuf_path)
+                module = importlib.import_module(module_name)
+                return module
+
+        def protos_and_services(protobuf_path, *, include_paths=None):
+            """Loads gRPC services and Protobuf messages from a .proto file
+
+            THIS IS AN EXPERIMENTAL API.
+
+            Equivalent to
+            ```
+            return google.protobuf.protos(protobuf_path, include_paths=include_paths),
+                    grpc.services(protobuf_path, include_paths=include_paths)
+            ```
+
+            Args:
+              protobuf_path: A string representing the path to the desired
+                ".proto" file.
+              include_paths: A sequence of strings which should be searched for
+                the ".proto" file. By default, the entries on sys.path are
+                searched.
+
+            Returns:
+              A 2-tuple of module objects. The first contains protocol buffer
+              message classes, while the other contains servicer classes.
+            """
+            protos_ = protobuf.protos(
+                protobuf_path, include_paths=include_paths)
+            services_ = services(protobuf_path, include_paths=include_paths)
+            return protos_, services_
+
+        sys.meta_path.extend([ProtoFinder()])
```

Comment:
How do we limit the `ProtoFinder` to be only effective within this file? Or why do we don't need to worry about it?

Reply:
It's not possible. The only way to implement a module loader is for the entire process. This *was* possible using Python 2's module loading infrastructure, but it's no longer possible in Python 3. If we're worried about people who don't want this Finder in their process, we can add an environment variable to turn the feature off.

Reply:
I think a comment about the Finder in the docstring is fine. Environment variable might be too burdensome, we can defer it until someone asking for it.